File: ios/Bulb.m
//
//  Bulb.m
//  RNLightViewApp
//
//  Created by Viet Pham on 2020-08-24.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(Bulb, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(isOn, BOOL)

RCT_EXPORT_VIEW_PROPERTY(onStatusChange, RCTDirectEventBlock)

@end

File: ios/Bulb.swift
//
//  Bulb.swift
//  RNLightViewApp
//
//  Created by Viet Pham on 2020-08-24.
//

import Foundation

@objc(Bulb)

class Bulb: RCTViewManager {
  
  override func view() -> UIView! {
    BulbView()
  }
  
  override class func requiresMainQueueSetup() -> Bool {
    return true
  }
  
}
